mvcc/backend/backend.go

```diff
@@ -179,15 +179,17 @@ func (b *backend) Size() int64 {
 
 func (b *backend) run() {
 	defer close(b.donec)
-
+	t := time.NewTimer(b.batchInterval)
+	defer t.Stop()
 	for {
 		select {
-		case <-time.After(b.batchInterval):
+		case <-t.C:
 		case <-b.stopc:
 			b.batchTx.CommitAndStop()
 			return
 		}
 		b.batchTx.Commit()
+		t.Reset(b.batchInterval)
```

Comment:
If you reset a timer, you have to handle the return value and read from t.C on true, otherwise you may get unpredictable ticks.  For this loop it may not matter now, but in the future it might.